### ISSUE TITLE
Get-azVM:  ‘this.client.subscriptionid’ cannot be null  

### ISSUE THREAD

**Issue by MikeWilcoxMicrosoft** [MicrosoftLearning/AZ-300-MicrosoftAzureArchitectTechnologies#166]
In the identity lab we run: 
From the az3000501-LabRG - Access control (IAM) blade, assign the Contributor role to the user-assigned managed identity az3000501-mi.

Switch back to the Remote Desktop session, and, from the PowerShell prompt, run the following to attempt to retrieve the currently used managed identity:

(Get-AzVM -ResourceGroupName az3000501-LabRG -Name az3000501-vm).Identity

And we receive the error: 
Get-AzVM : No subscription found in the context.  Please ensure that the credentials you provided are authorized to
access an Azure subscription, then run Connect-AzAccount to login.
At line:1 char:2
+ (Get-AzVM -ResourceGroupName az3000501-LabRG -Name az3000501-vm).Iden ...
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Get-AzVM], ApplicationException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Compute.GetAzureVMCommand

**Comment by MikeWilcoxMicrosoft** [comment]
more info.. Where is the Subscription Name ? 

PS C:\Users\Student> Add-AzAccount -Identity


Account   SubscriptionName TenantId                             Environment
-------   ---------------- --------                             -----------
MSI@50342                  674ce2a1-db99-41c8-95c4-529b43ff69db AzureCloud

**Comment by polichtm** [comment]
Hi Mike,
pls provide the lab, exercise, taks, and step number
rgds
Marcin

**Comment by MikeWilcoxMicrosoft** [comment]
Task 2 , line 2 
in 
  https://github.com/MicrosoftLearning/AZ-300-MicrosoftAzureArchitectTechnologies/blob/master/Instructions/AZ-300T01_Lab_Mod05_Implementing%20user-assigned%20managed%20identities%20for%20Azure%20resources.md

**Comment by MikeWilcoxMicrosoft** [comment]
![image](https://user-images.githubusercontent.com/57420850/84778820-2303b200-afa9-11ea-9c79-1d79c50fa603.png)
Here is a screen shot.. thank you

**Comment by MikeWilcoxMicrosoft** [comment]
I noticed that the start menu was not available in the Windows2016 VM used by the lab ?  Is this normal?  Separately where do i find the Administrator password ?

**Comment by MikeWilcoxMicrosoft** [comment]
hmm one of my class team mates said I should run the power shell in the command shell to run the commands and not try to run them on the vm .. Thoughts ?

**Comment by djcmct** [comment]
> I noticed that the start menu was not available in the Windows2016 VM used by the lab ? Is this normal? Separately where do i find the Administrator password ?

Hi Mike, the lab deploys Windows Server Core version, so no GUI or start menu and the password is in the azuredeploy05.parameters.json uploaded in the earlier exercise to deploy the VM

**Comment by MikeWilcoxMicrosoft** [comment]
[Managing Identities Implementing user-assigned managed identities for Azure resources Lab Managed 300T01_Lab_Mod05 Identity screen shots.docx](https://github.com/MicrosoftLearning/AZ-300-MicrosoftAzureArchitectTechnologies/files/4787870/Managing.Identities.Implementing.user-assigned.managed.identities.for.Azure.resources.Lab.Managed.300T01_Lab_Mod05.Identity.screen.shots.docx)

**Comment by djcmct** [comment]
> hmm one of my class team mates said I should run the power shell in the command shell to run the commands and not try to run them on the vm .. Thoughts ?

Running the commands from PS within the VM is correct, the idea behind the lab is to demonstrate that MI's can be used as service accounts, so getting the VM to call the managed identity assigned to it to perform the actions against the roles allocated allows the VM to use the MI to complete the tasks required by elevating to that identity rather than having to embed a username/password into the process. Not sure why the sub id is missing, thats not correct, is probably worth deleting everything and starting from scratch, i delivered this last week to a group of 20 and all worked ok

**Comment by MikeWilcoxMicrosoft** [comment]
thank you.. I started from scratch and there are some vague tasks for newbies, so maybe my screen shots will help others..

**Comment by MikeWilcoxMicrosoft** [comment]
Marcin,
I updated the issue with more info.
Unless something changed, it worked today..  ie maybe ‘user error’ ☹

Here is the issue # and btw I included screen shots that might help other lab students.

https://github.com/MicrosoftLearning/AZ-300-MicrosoftAzureArchitectTechnologies/issues/166

Mike Wilcox
Industry Technology Strategist
Cloud Solution Architect

Mobile:   +1 (972) 786-3978
Mike.Wilcox@microsoft.com<mailto:Mike.Wilcox@microsoft.com>

http://www.linkedin.com/in/mwilcox1<https://nam06.safelinks.protection.outlook.com/?url=http%3A%2F%2Fwww.linkedin.com%2Fin%2Fmwilcox1&data=02%7C01%7CMike.Wilcox%40microsoft.com%7C0cf59d1ab82844a622cf08d7fa881ac1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637253334774691684&sdata=QPNz1u5hDPreOwby6BZ0GamYgP3pduif2XLcLY5N240%3D&reserved=0>




[MSFT_logo_Gray DE sized SIG1.png]

[linked.png]<http://www.linkedin.com/in/mwilcox1>   [cid:image003.png@01D643F7.52E43940] <http://ieee.org/>

**Comment by MikeWilcoxMicrosoft** [comment]
update:  I ran into the same issues today where in powershell,this time on my laptop the subscription names did not exist.  I opened a stack overflow to see if others have any ideas. 
https://stackoverflow.com/questions/62418089/this-client-subscriptionid-cannot-be-null

**Comment by polichtm** [comment]
Hi Mike - unfortunately I cannot reproduce he issue - let us know what you find out
rgds
Marin

**Comment by MikeWilcoxMicrosoft** [comment]
I have tried both Azure Powershell  v5 and v7 on my windows 10 laptop and when I run:
> Get-AzContext -ListAvailable

Name                                     Account             SubscriptionName    Environment         TenantId
----                                     -------             ----------------    -----------         --------
674ce2a1-db99-41c8 - … zoo@microsoft…                                       AzureCloud          674ce2a1..  

Note, the SubscriptionName is still empty.  Basically I can not run any Azure Powershell commands in my laptop Powershell.   
Note, if i run the same command in the portal 'command shell' the SubscriptionName is returned correctly.

**Comment by polichtm** [comment]
Hi Mike
pls clarify which exercise, task, and step you are referring to
rgds
Marcin

**Comment by MikeWilcoxMicrosoft** [comment]
Sorry for the confusion but this problem is for ALL powershell commands in ALL exercises, labs and any usage of Azure Powershell.

**Comment by polichtm** [comment]
Hi Mike,
Apologies but I'm not following. Are you looking for a general help with your Azure subscription?
rgds
Marcin

**Comment by hawkism** [comment]
Mike, try some of the below - 
This is what I do to get started with Azure and powershell commands

Connect-AzAccount
Get-AzSubscription

Set-AzContext -Subscription <subscription ID here> -Name “<what you want the subscription to be called, here>“

Select-AzContext “<what you set the subscription name to, above>”

The quotes need to stay -
 
Give the above a try.  I had to struggle through this with powershell at first. 

Initially my AzContect Name was really long  - it had the subscription name, subscription ID, and my email address all together, until I renamed it. 




Steven M. Hawkins
